File: src/PortfolioTracker.Ui/src/app/models/asset.ts
import { AssetType } from '../enums/assetType';
import { Currency } from '../enums/currency';
import { RiskLevel } from '../enums/riskLevel';

export interface IAsset {
  id: string;
  accountId: string;
  name: string;
  assetType: AssetType;
  exchangeTicker: string | null;
  openPrice: number | null;
  interestRate: number | null;
  currency: Currency;
  units: number;
  riskLevel: RiskLevel;
}

File: src/PortfolioTracker.Domain/Models/Asset.cs
using System;
using PortfolioTracker.Events.Common;

namespace PortfolioTracker.Domain.Models
{
    public class Asset
    {
        public string Id { get; set; } = null!;
        public string AccountId { get; set; } = null!;
        public string UserId { get; set; } = null!;
        public string Name { get; set; } = null!;
        public AssetType AssetType { get; set; }
        public string? ExchangeTicker { get; set; }
        public decimal? OpenPrice { get; set; }
        public double? InterestRate { get; set; }
        public decimal Units { get; set; }
        public Currency Currency { get; set; }
        public RiskLevel RiskLevel { get; set; }
        public DateTimeOffset CreatedAt { get; set; }
    }
}
